### ISSUE TITLE
MinGW/GCC 4.9 compilation errors

### ISSUE THREAD

**Issue by johngull** [DGtal-team/DGtal#1204]
As you know M_PI, M_PI_4 and so on is not defined now in C++ standard.
But it is used in DGtal, for example in BasicFunctors.h and it leads to compilation errors.

My current fix to use you library is to add:
#ifndef M_PI
#define M_PI (3.14159265358979323846)
#endif

But I would recommend to use boost constants from here: boost/math/constants/constants.hpp

**Comment by dcoeurjo** [comment]
Thanks a lot for the report. 

In Common.h (included in BasicFunctors), we have a 
````c++
#if defined( WIN32 )
#define _USE_MATH_DEFINES
#include <math.h>
#else
#include <cmath>
#endif //win32
````
that is supposed to define the M_PI on most platforms.

Would 
````c++
#define M_PI (boost::math::constants::pi())
````
be really safe and efficient (I don't know if boost constants are implemented as constexpr for instance) ?

**Comment by rolanddenis** [comment]
The templated version of boost constants (when using for instance `boost::math::constants::pi<double>()`) is not `constexpr` before `1.50.0`. It starts with `1.58.0` for the non-templated versions (like `boost::math::double_constants::pi`). Since Travis uses `1.46.1`, I don't see other solution than defining it the hard way when `M_PI` isn't already defined, like proposed by @johngull. Actually, it is already the case in https://github.com/DGtal-team/DGtal/blob/master/src/Board/Shapes.h#L44.

Nevertheless, I agree that it would be cleaner to use `boost::math::constants::pi<T>()` (or a similar version shipped with DGtal) when needed instead of relying on a macro.

**Comment by dcoeurjo** [comment]
Fixed in #1208 , thanks for the report.
